python/openinference-instrumentation/src/openinference/instrumentation/__init__.py

```diff
@@ -12,11 +33,147 @@ def suppress_tracing() -> Iterator[None]:
             # No tracing will occur within this block
             ...
     """
-    try:
-        from opentelemetry.context import _SUPPRESS_INSTRUMENTATION_KEY, attach, detach, set_value
-    except ImportError:
-        yield
+
+    def __enter__(self) -> "suppress_tracing":
+        self._token = attach(set_value(_SUPPRESS_INSTRUMENTATION_KEY, True))
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback) -> None:
+        """Raise any exception triggered within the runtime context."""
+        detach(self._token)
+        return None
+
+
+class UsingAttributes:
+    slots = [
+        "_token",
+        "_session_id",
+        "_user_id",
+        "_metadata",
+        "_tags",
+    ]
+
+    def __init__(
+        self,
+        session_id: str = "",
+        user_id: str = "",
+        metadata: Dict[str, str] = {},
+        tags: List[str] = [],
```

Comment:
These need to be optional types, because the user should have the option to not set them. Might make sense to add these individually in follow-up PRs.

Reply:
I'd also note it's not safe to use mutable types such as dictionaries and lists as default arguments in Python.

https://www.pullrequest.com/blog/python-pitfalls-the-perils-of-using-lists-and-dicts-as-default-arguments/#:~:text=However%2C%20the%20problem%20lies%20with,t%20provide%20an%20item%20list.

Reply:
Made the input `Optional` and default to `None`. Thanks!